src/wallet/rpcwallet.cpp

```diff
@@ -305,6 +305,289 @@ UniValue sendtoaddress(const UniValue& params, bool fHelp)
+        }
+    }
+
+    /// imported_watchonly source
+    {
+        UniValue entry(UniValue::VOBJ);
+        entry.pushKV("source", "imported_watchonly");
+        bool hasData = false;
+
+        if (!t_watchonly_dests.empty()) {
+            UniValue watchonly_t_addrs(UniValue::VARR);
+            for (const CTxDestination& dest: t_watchonly_dests) {
+                watchonly_t_addrs.push_back(keyIO.EncodeDestination(dest));
+            }
+
+            UniValue watchonly_t(UniValue::VOBJ);
+            watchonly_t.pushKV("addresses", watchonly_t_addrs);
+
+            entry.pushKV("transparent", watchonly_t);
+            hasData = true;
+        }
+
+        {
+            UniValue watchonly_sprout_addrs(UniValue::VARR);
+            for (const SproutPaymentAddress& addr : sproutAddresses) {
+                if (!HaveSpendingKeyForPaymentAddress(pwalletMain)(addr)) {
+                    watchonly_sprout_addrs.push_back(keyIO.EncodePaymentAddress(addr));
+                }
+            }
+
+            if (!watchonly_sprout_addrs.empty()) {
+                UniValue watchonly_sprout(UniValue::VOBJ);
+                watchonly_sprout.pushKV("addresses", watchonly_sprout_addrs);
+                entry.pushKV("sprout", watchonly_sprout);
+                hasData = true;
+            }
+        }
+
+        {
+            UniValue watchonly_sapling_addrs(UniValue::VARR);
+            for (const SaplingPaymentAddress& addr : saplingAddresses) {
+                if (!HaveSpendingKeyForPaymentAddress(pwalletMain)(addr)) {
+                    watchonly_sapling_addrs.push_back(keyIO.EncodePaymentAddress(addr));
+                }
+            }
+
+            if (!watchonly_sapling_addrs.empty()) {
+                UniValue watchonly_sapling_obj(UniValue::VOBJ);
+                watchonly_sapling_obj.pushKV("addresses", watchonly_sapling_addrs);
+                UniValue watchonly_sapling(UniValue::VARR);
+                watchonly_sapling.push_back(watchonly_sapling_obj);
+
+                entry.pushKV("sapling", watchonly_sapling);
+                hasData = true;
+            }
+        }
+
+        if (hasData) {
+            ret.push_back(entry);
+        }
+    }
+
+    /// legacy_hdseed source
+    {
+        UniValue entry(UniValue::VOBJ);
+        entry.pushKV("source", "legacy_hdseed");
+        bool hasData = false;
+
+        std::map<SaplingIncomingViewingKey, std::vector<SaplingPaymentAddress>> ivkAddrs;
+        for (const SaplingPaymentAddress& addr : saplingAddresses) {
+            if (GetSourceForPaymentAddress(pwalletMain)(addr) == LegacyHDSeed) {
+                SaplingIncomingViewingKey ivkRet;
+                if (pwalletMain->GetSaplingIncomingViewingKey(addr, ivkRet)) {
+                    ivkAddrs[ivkRet].push_back(addr);
+                }
+            }
+        }
+
+        {
+            UniValue legacy_sapling(UniValue::VARR);
+            for (const auto& [ivk, addrs] : ivkAddrs) {
```

Comment:
Is this C++17 syntax? Huh.

Reply:
It's actually C++11. I discovered it in the example code on the std::map documentation page.